system-parachains/collectives/collectives-polkadot/src/impls.rs

```diff
@@ -38,36 +39,44 @@ pub type BalanceOf<T> =
 
 /// Implements `OnUnbalanced::on_unbalanced` to teleport slashed assets to relay chain treasury
 /// account.
-pub struct ToParentTreasury<TreasuryAccount, PalletAccount, T>(
-	PhantomData<(TreasuryAccount, PalletAccount, T)>,
+pub struct ToParentTreasury<TreasuryAccount, AccountIdConverter, T>(
+	PhantomData<(TreasuryAccount, AccountIdConverter, T)>,
 );
 
-impl<TreasuryAccount, PalletAccount, T> OnUnbalanced<NegativeImbalance<T>>
-	for ToParentTreasury<TreasuryAccount, PalletAccount, T>
+impl<TreasuryAccount, AccountIdConverter, T> OnUnbalanced<NegativeImbalance<T>>
+	for ToParentTreasury<TreasuryAccount, AccountIdConverter, T>
 where
 	T: pallet_balances::Config + pallet_xcm::Config + frame_system::Config,
 	<<T as frame_system::Config>::RuntimeOrigin as OriginTrait>::AccountId: From<AccountIdOf<T>>,
 	[u8; 32]: From<<T as frame_system::Config>::AccountId>,
 	TreasuryAccount: Get<AccountIdOf<T>>,
-	PalletAccount: Get<AccountIdOf<T>>,
 	BalanceOf<T>: Into<Fungibility>,
+	AccountIdConverter: ConvertLocation<AccountIdOf<T>>,
 {
 	fn on_unbalanced(amount: NegativeImbalance<T>) {
 		let amount = match amount.drop_zero() {
 			Ok(..) => return,
 			Err(amount) => amount,
 		};
 		let imbalance = amount.peek();
-		let pallet_acc: AccountIdOf<T> = PalletAccount::get();
-		let treasury_acc: AccountIdOf<T> = TreasuryAccount::get();
-
-		<pallet_balances::Pallet<T>>::resolve_creating(&pallet_acc, amount);
+		let root_location: MultiLocation = Here.into();
```

Comment:
there are few options, 
- (current) resolve funds into root account temporarily and teleport on behalf of the root 
- build custom xcm program to teleport (do not use xcm pallet)
- keep as it is, and grant free execution to the pallet's account id. where a location of an account id has to be listed as `X1(AccountId32(some_custom_id))`

I think the current is the most optimal